src/clib/pio_nc.c

```diff
@@ -2800,15 +2834,19 @@ int PIOc_def_dim(int ncid, const char *name, PIO_Offset len, int *idp)
         file->dim_values[file->num_dim_vars] = len;
         *idp = file->num_dim_vars;
         ++file->num_dim_vars;
-        adios2_error adiosErr = adios2_put(file->engineH, variableH, &len, adios2_mode_sync);
-        file->num_written_blocks += file->num_all_procs;
-        if (adiosErr != adios2_error_none)
+        adios2_error adiosErr = adios2_error_none;
+        if (file->adios_io_process == 1)
         {
-            spio_ltimer_stop(ios->io_fstats->tot_timer_name);
-            spio_ltimer_stop(file->io_fstats->tot_timer_name);
-            return pio_err(ios, file, PIO_EADIOS2ERR, __FILE__, __LINE__,
-                           "Putting (ADIOS) variable (name=%s) failed (adios2_error=%s) for file (%s, ncid=%d)",
-                           dimname, convert_adios2_error_to_string(adiosErr), pio_get_fname_from_file(file), file->pio_ncid);
+            adiosErr = adios2_put(file->engineH, variableH, &len, adios2_mode_sync);
+            file->num_written_blocks += file->num_alltasks;
```

Comment:
Should this line be after the error checking (even if put fails do you want to update the blocks written out)?

Reply:
Yes, it can be moved after the error check.